x-pack/plugins/enterprise_search/public/applications/app_search/components/credentials/credentials_list/credentials_list.test.tsx

```diff
@@ -0,0 +1,276 @@
+          ],
+        },
+      });
+      const wrapper = shallow(<CredentialsList />);
+      const { items } = wrapper.find(EuiBasicTable).props();
+      expect(items.map((i: IApiToken) => i.id)).toEqual([undefined, 1, 2]);
+    });
+  });
+
+  describe('pagination', () => {
+    it('derives pagination from meta object', () => {
+      mockKea({
+        values: {
+          meta: {
+            page: {
+              current: 6,
+              size: 55,
+              total_pages: 1,
+              total_results: 1004,
+            },
+          },
+        },
+      });
+
+      const wrapper = shallow(<CredentialsList />);
+      const { pagination } = wrapper.find(EuiBasicTable).props();
+      expect(pagination).toEqual({
+        pageIndex: 5,
+        pageSize: 55,
+        totalItemCount: 1004,
+        hidePerPageOptions: true,
+      });
+    });
+
+    it('will default pagination values if `page` is not available', () => {
+      mockKea({
+        values: {
+          meta: {},
+        },
+      });
+
+      const wrapper = shallow(<CredentialsList />);
+      const { pagination } = wrapper.find(EuiBasicTable).props();
+      expect(pagination).toEqual({
+        pageIndex: 0,
+        pageSize: 0,
+        totalItemCount: 0,
+        hidePerPageOptions: true,
+      });
+    });
+  });
+
+  describe('columns', () => {
+    let columns: Array<{
+      render: (token: object) => any;
+      actions: any;
+    }>;
+    const showCredentialsForm = jest.fn();
+    const deleteApiKey = jest.fn();
+
+    beforeEach(() => {
+      mockKea({
+        actions: {
+          showCredentialsForm,
+          deleteApiKey,
+        },
+      });
+      const wrapper = shallow(<CredentialsList />);
+      columns = wrapper.find(EuiBasicTable).props().columns;
+    });
+
+    describe('column 1 (name)', () => {
+      const token = {
+        ...apiToken,
+        name: 'some-name',
+      };
+
+      it('renders correctly', () => {
+        const column = columns[0];
+        const wrapper = shallow(<div>{column.render(token)}</div>);
+        expect(wrapper.text()).toEqual('some-name');
+      });
+    });
+
+    describe('column 2 (type)', () => {
+      const token = {
+        ...apiToken,
+        type: ApiTokenTypes.Private,
+      };
+
+      it('renders correctly', () => {
+        const column = columns[1];
+        const wrapper = shallow(<div>{column.render(token)}</div>);
+        expect(wrapper.text()).toEqual('Private API Key');
+      });
+    });
+
+    describe('column 3 (key)', () => {
+      const testToken = {
+        ...apiToken,
+        key: 'abc-123',
+      };
+
+      const subject = (token: object): [any, any] => {
+        const copyMock = jest.fn();
+        const column = columns[2];
+        const wrapper = shallow(<div>{column.render(token)}</div>);
+        const children = wrapper.find(EuiCopy).props().children;
+        const copyEl = shallow(children(copyMock));
+        return [copyMock, copyEl];
+      };
```

Comment:
Ooo, this is super cool. Very nice!

Reply:
What is?

Reply:
The subject() helper - I like it a lot, we might want to DRY it out in the future if we have any more components that use EuiCopy (although I think credentials might be the only one, so nevermind haha).